designer/datasource-editor-jdbc/src/main/java/org/pentaho/reporting/ui/datasources/jdbc/ui/JdbcPreviewWorker.java

```diff
@@ -88,19 +108,19 @@ public void cancelProcessing(final CancelEvent event)
    *
    * @see Thread#run()
    */
-  public void run()
-  {
-    try
-    {
-      tableModel = dataFactory.queryData
-          (query, new QueryDataRowWrapper(new ParameterDataRow(), queryLimit, queryTimeout));
-      if (queryLimit > 0)
-      {
-        tableModel = new LengthLimitingTableModel(tableModel, queryLimit);
+  public void run() {
+    try {
+      ParameterDataRow paramDataRow = new ParameterDataRow();
+      if ( this.reportParameters != null ) {
+        paramDataRow = normalizeParameters( this.reportParameters );
       }
-    }
-    catch (ReportDataFactoryException e)
-    {
+
+      tableModel = dataFactory.queryData( query, new QueryDataRowWrapper( paramDataRow, queryLimit, queryTimeout ) );
+
+      if ( queryLimit > 0 ) {
```

Comment:
This should be greater or equal to 0. A 0 limit must also be enforced.

Reply:
Nope. We are using JDBC semantics, therefore zero means no limit. See https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#setMaxRows(int)

Reply:
I hadn't change anything there @lucboudreau. The code changes that I did were:
JdbcDataSourceDialog - Line 136 to 143 - Getting report parameters
JdbcPreviewWorker - Most code changes - Receiving parameters, treat them and pass to query treatment flow
The other java class (JdbcQueryDesignDialog l.55) is a side effect of function signature change.